libraries/app/database_api.cpp

```diff
@@ -2070,29 +2070,27 @@ bool database_api::verify_account_authority( const string& account_name_or_id, c
    return my->verify_account_authority( account_name_or_id, signers );
 }
 
-bool database_api_impl::public_key_found(const flat_set<public_key_type>& to_be_found,
-      const vector<public_key_type>& collection_to_search) const
-{
-   for (public_key_type from_collection : collection_to_search)
-   {
-      for(public_key_type passed_in : to_be_found ) 
-      {
-         if (passed_in == from_collection)
-         {
-            return true;
-         }
-      }
-   }
-   return false;
-}
-
 bool database_api_impl::verify_account_authority( const string& account_name_or_id, 
       const flat_set<public_key_type>& keys )const
 {
-   const account_object* account = get_account_from_string(account_name_or_id);
+   // create a dummy transfer
+   transfer_operation op;
+   op.from = get_account_from_string(account_name_or_id)->id;
+   std::vector<operation> ops;
+   ops.emplace_back(op);
+
+   try
```

Comment:
why not just to create `signed_transaction` and call it's `verify_authority` method ?
So the full implementation of `verify_account_authority` might be like next:
```
bool database_api_impl::verify_account_authority( const string& account_name_or_id, 
      const flat_set<public_key_type>& keys )const
{
   // create a dummy transfer
   transfer_operation op;
   op.from = get_account_from_string(account_name_or_id)->id;
   std::vector<operation> ops;
   ops.emplace_back(op);
   signed_transaction trx;
   trx.signees = keys;
   trx.operations.push_back(op);

   return verify_authority(trx);
}
```
It's less code and more simple IMHO.

Reply:
Nice idea, but IMO the signees field should be used as an internal cache only, not part of the tx struct's interface.

Reply:
`graphene::chain::verify_authority` will use `signees` returned by `signed_transaction::get_signature_keys` method, so it should work